core_gl/CMakeLists.txt

```diff
@@ -22,7 +22,7 @@ if(BUILD_CORE)
   file(GLOB_RECURSE shader_files RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "Shaders/*")
 
   # Target definition
-  add_library(${PROJECT_NAME} STATIC ${public_header_files} ${header_files} ${shader_files} ${source_files})
+  add_library(${PROJECT_NAME} OBJECT ${public_header_files} ${header_files} ${shader_files} ${source_files})
```

Comment:
When making the core_gl an object library, probably the core_gl objects can be attached to the core lib itself when gl is enabled.